lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py

```diff
@@ -0,0 +1,404 @@
+    file_manager = content.guestOperationsManager.fileManager
+
+    try:
+        fileTransferInfo = file_manager.InitiateFileTransferFromGuest(vm=vm, auth=creds,
+                                                                      guestFilePath=src)
+    except vim.fault.FileNotFound:
+        result['changed'] = False
+        result['msg'] = "Guest file %s not exists" % src
+        return result
+    except vim.fault.FileFault as e:
+        module.fail_json(msg="FileFault:%s" % e.msg, uuid=vm.summary.config.uuid)
+    except vim.fault.GuestPermissionDenied:
+        module.fail_json(msg="Permission denied to fetch file %s" % src, uuid=vm.summary.config.uuid)
+    except vim.fault.InvalidGuestLogin:
+        module.fail_json(msg="Invalid guest login for user %s" % vm_username, uuid=vm.summary.config.uuid)
+    # other exceptions
+    except Exception as e:
+        module.fail_json(msg="Failed to Fetch file from Vm VMware exception:%s" % e, uuid=vm.summary.config.uuid)
+
+    url = fileTransferInfo.url
+    try:
+        resp, info = urls.fetch_url(module, url, method="GET")
+    except Exception as e:
+        module.fail_json(msg="Failed to Fetch file from Vm VMware exception:%s" % e,
+                         uuid=vm.summary.config.uuid)
+
+    try:
+        with open(dest, "wb") as local_file:
+            local_file.write(resp.read())
+    except Exception as e:
+        module.fail_json(msg="local file write exception:%s" % e, uuid=vm.summary.config.uuid)
+
+    return result
+
+
+def copy(module, content, vm):
+    result = {}
+    result['changed'] = True
+    result['uuid'] = vm.summary.config.uuid
+    vm_username = module.params['vm_username']
+    vm_password = module.params['vm_password']
+    overwrite = module.params["copy"]["overwrite"]
+    dest = module.params["copy"]['dest']
+    src = module.params['copy']['src']
+    b_src = to_bytes(src, errors='surrogate_or_strict')
+
+    if not os.path.exists(b_src):
+        module.fail_json(msg="Source %s not found" % (src))
+    if not os.access(b_src, os.R_OK):
+        module.fail_json(msg="Source %s not readable" % (src))
+    if os.path.isdir(b_src):
+        module.fail_json(msg="copy does not support copy of directory: %s" % (src))
+
+    data = None
+    with open(b_src, "r") as local_file:
+        data = local_file.read()
+    file_size = os.path.getsize(b_src)
+
+    creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+    file_attributes = vim.vm.guest.FileManager.FileAttributes()
+    file_manager = content.guestOperationsManager.fileManager
+    try:
+        url = file_manager.InitiateFileTransferToGuest(vm=vm, auth=creds, guestFilePath=dest,
+                                                       fileAttributes=file_attributes, overwrite=overwrite,
+                                                       fileSize=file_size)
+    except vim.fault.FileAlreadyExists:
```

Comment:
Are you sure `vim.fault.FileAlreadyExists` can be raised by `InitiateFileTransferToGuest`, it is [not listed in the doc](https://github.com/vmware/pyvmomi/blob/master/docs/vim/vm/guest/FileManager.rst) ?